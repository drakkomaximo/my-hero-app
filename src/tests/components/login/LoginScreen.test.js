import { mount } from "enzyme"
import { AuthContext } from "../../../auth/AuthContex"
import LoginScreen from "../../../components/login/LoginScreen"
import { types } from "../../../types/types"

describe('Tests on <LoginScreen />', () => {

    const historyMock ={
        replace: jest.fn()
    }

    const contextValue = {
        dispatch: jest.fn()
    }

    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={historyMock}/>
        </AuthContext.Provider>
    )

    test('It must show correctly', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('It must do the dispatch and the navigate', () => {
        const handelClick = wrapper.find('button').prop('onClick')

        handelClick()

        expect( contextValue.dispatch ).toHaveBeenLastCalledWith({
            type: types.login,
            payload:{
                name:'Andr√©s'
            }
        })

        expect( historyMock.replace ).toHaveBeenCalledWith('/')

        localStorage.setItem('lastPath', '/dc')

        handelClick()

        expect( historyMock.replace ).toHaveBeenCalledWith('/dc')

    })


    
    
})
