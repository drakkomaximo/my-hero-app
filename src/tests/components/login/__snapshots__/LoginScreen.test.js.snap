// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <LoginScreen /> It must show correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
